rust-symcrypt/src/rsa/mod.rs

```diff
@@ -0,0 +1,716 @@
+            let result = symcrypt_sys::SymCryptRsakeyGetValue(
+                self.inner.0,
+                modulus_buffer.as_mut_ptr(),
+                modulus_buffer.len() as symcrypt_sys::SIZE_T,
+                pub_exp.as_mut_ptr(),
+                1 as symcrypt_sys::UINT32,
+                [p.as_mut_ptr(), q.as_mut_ptr()].as_mut_ptr(),
+                primes_len.as_mut_ptr(),
+                2 as symcrypt_sys::UINT32,
+                NumberFormat::MSB.to_symcrypt_format(),
+                self.key_usage.to_symcrypt_flag(),
+            );
+            if result != symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR {
+                return Err(result.into());
+            }
+
+            let result = symcrypt_sys::SymCryptRsakeyGetCrtValue(
+                self.inner.0,
+                [d_p.as_mut_ptr(), d_q.as_mut_ptr()].as_mut_ptr(),
+                crt_lens.as_mut_ptr(),
+                2,
+                crt_coefficient.as_mut_ptr(),
+                crt_coefficient.len() as symcrypt_sys::SIZE_T,
+                private_exponent.as_mut_ptr(),
+                private_exponent.len() as symcrypt_sys::SIZE_T,
+                NumberFormat::MSB.to_symcrypt_format(),
+                self.key_usage.to_symcrypt_flag(),
+            );
+            if result != symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR {
+                return Err(result.into());
+            }
+        }
+
+        let pub_exp_bytes = store_msb_first_u64(pub_exp[0], self.get_size_of_public_exponent())?;
+
+        Ok(RsaKeyPairExportBlob {
+            modulus: modulus_buffer,
+            pub_exp: pub_exp_bytes,
+            p,
+            q,
+            d_p,
+            d_q,
+            crt_coefficient,
+            private_exp: private_exponent,
+        })
+    }
+
+    /// `export_public_key_blob()` will export a [`RsaPublicKeyExportBlob`].
+    pub fn export_public_key_blob(&self) -> Result<RsaPublicKeyExportBlob, SymCryptError> {
+        let mut modulus_buffer = vec![0u8; self.get_size_of_modulus() as usize];
+        let mut pub_exp = vec![0u64; 1];
+        unsafe {
+            // SAFETY: FFI calls
+            // When only getting the public key, ppPrimes, pcbPrimes and nPrimes can be NULL, NULL and 0.
+            match symcrypt_sys::SymCryptRsakeyGetValue(
+                self.inner.0,
+                modulus_buffer.as_mut_ptr(),
+                modulus_buffer.len() as symcrypt_sys::SIZE_T,
+                pub_exp.as_mut_ptr(),
+                1 as symcrypt_sys::UINT32,
+                ptr::null_mut(),
+                ptr::null_mut(),
+                0,
+                NumberFormat::MSB.to_symcrypt_format(),
+                self.key_usage.to_symcrypt_flag(),
+            ) {
+                symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR => Ok(RsaPublicKeyExportBlob {
+                    modulus: modulus_buffer,
+                    pub_exp: store_msb_first_u64(pub_exp[0], self.get_size_of_public_exponent())?,
```

Comment:
If you return the pub_exp as a u64 value, then, you will need to provide a public function to convert to multi-byte array and document.